src/software/thunderscope/binary_context_managers.py

```diff
@@ -452,16 +458,18 @@ def next_free_port(self, port=40000, max_port=65535):
         raise IOError("no free ports")
 
     def setup_proto_unix_io(
-        self, blue_full_system_proto_unix_io, yellow_full_system_proto_unix_io
-    ):
+        self,
+        blue_full_system_proto_unix_io: ProtoUnixIO,
+        yellow_full_system_proto_unix_io: ProtoUnixIO,
+    ) -> None:
         """Setup gamecontroller io
 
         :param blue_full_system_proto_unix_io: The proto unix io of the blue full system.
         :param yellow_full_system_proto_unix_io: The proto unix io of the yellow full system.
 
         """
 
-        def __send_referee_command(data):
+        def __send_referee_command(data: Message) -> None:
```

Comment:
nit: `Message` is the generic abstract class that all protobuf messages extend. In cases like this we can be more specific
```suggestion
        def __send_referee_command(data: Referee) -> None:
```